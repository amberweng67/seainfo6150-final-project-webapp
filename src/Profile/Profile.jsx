import React from 'react';
import styles from './Profile.module.css';
import CoachCard from '../CoachCard/CoachCard';
import coachMedium from '../assets/coach-400.webp';
import coachLarge from '../assets/coach-600.webp';

const Profile = ({ user, coachList }) => {
  return (
    <div className={styles.container}>
      <div className={styles.imageContainer}>
        <img
          alt="coach"
          src={coachLarge}
          srcSet={`${coachMedium} 400w, ${coachLarge} 600w`} sizes="(max-width: 960px) 400px, 600px" />
      </div>
      <h3>{user}</h3>
      <h4>Coach list:</h4>
      <div className={styles.listContainer}>
        {coachList.map(card => <CoachCard key={card.id} card={card} />)}
      </div>
    </div>
  );
}

export default Profile;