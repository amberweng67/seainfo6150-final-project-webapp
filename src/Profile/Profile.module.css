.container {
  padding: 64px;
}

.listContainer {
  display: flex;
  flex-wrap: wrap;
}

.listContainer > *:not(:last-child) {
  margin-right: 16px;
  margin-bottom: 16px;
}

.imageContainer {
  display: flex;
  justify-content: center;
}