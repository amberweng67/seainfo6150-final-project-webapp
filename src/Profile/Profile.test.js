import React from "react";
import Profile from "./Profile.jsx";

it("renders the profile component correctly", () => {
  const { container } = render(<Profile user={'test'} coachList={[]}/>);
  expect(container).toMatchSnapshot();
});
