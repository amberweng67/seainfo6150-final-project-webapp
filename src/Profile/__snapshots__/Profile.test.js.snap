// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the profile component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h3>
      test
    </h3>
    <h4>
      Coach list:
    </h4>
    <div
      class="listContainer"
    />
  </div>
</div>
`;
